misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt

```diff
@@ -22,23 +23,26 @@ class ConfigMetadataAction @Inject constructor(
   config: Config,
   private val jvmMetadataAction: JvmMetadataAction,
   private val mode: ConfigTabMode,
+  @ConfigRedactKey keysToRedact: List<String>
 ) : WebAction {
+  private fun redact(
+    mapOfOutputs: Map<String, String?>,
+    regex: Regex
+  ) = mapOfOutputs
+    .mapValues { it.value?.replace(regex, "████████") }
+
+  // TODO make this config variable specific redaction with an annotation on config field, not just multibound list of keys
+  // Regex to match on redacted keys for redaction in output
+  private val yamlFilesRegex = Regex("(?<=(${keysToRedact.joinToString("|")}): )([^\n]*)")
```

Comment:
I know the shape of this pattern is existing, but it has effectively not been used for a while.

I think it's generally good idea to have unit tests specifically for regexes to verify they are behaving as we expect. At first glance, it would seem this matcher would be overly greedy if we had a key inside the value like

```
  some_key: "The quick brown fox found his password: it was 'what does the fox say?'"
```

becomes 

```
  some_key: "The quick brown fox found his password: ████████"
```

Though, I'd be more concerned about _missing_ something than being overly greedy for an edge case like that...

YAML does have [a lot of ways](https://stackoverflow.com/a/21699210) to write multi-line values which I think will defeat this pattern and evade redaction. Though, maybe not a concern in this flow if a trip through deserialization normalizes those to always work with this pattern?

Though, I think the premise of this whole key matching is a bit flawed...Anything that you include directly in your config _should not be a secret_ because it's checked in to source control, anybody in the company can download the deployable artifact, etc. And, sure enough, in general we see things like `passphrase: doesntmatter` in our configs.

Using regex here ultimately feels like the wrong layer to do what we need. Possibly worth thinking about redaction from the perspective of keys that _load_ their content from the filesystem/classpath and do the scrubbing a layer _above_ serialization?

And all of the above ^ has me continuing to wonder if we wouldn't just be better off biting the bullet of introducing a `!secret` directive in misk as well. Though I don't want to hold up useful config to wait for all existing secrets to move to a new structure like that...

Happy to chat/pair through some of this stuff f2f tomorrow.

Reply:
Thanks for the fuller thoughts there, I managed to get annotated redaction working instead which I think covers the effective config case sufficiently. Let me know if there's any case that we might need additional coverage for.